### PR TITLE
Tests are no longer part of the package

### LINKED ISSUE #6526: Move the tests directory outside of the code directory
### Discussed in https://github.com/pymc-devs/pymc/discussions/6172

<div type='discussions-op-text'>

<sup>Originally posted by **Armavica** October  2, 2022</sup>
Hi,
With the reorganization of the tests complete (#5777), a last question remains: where to place the `tests` directory. It currently resides inside the `pymc` code directory:

```
pymc/
├── benchmarks/
├── docs/
├── pymc/
│   ├── __init__.py
│   ├── model.py
│   ├── ...
│   └── tests/
└── scripts/
```

but one could argue that it would make more sense to put it at the same level as `benchmarks/`, `docs/` and `pymc/`:

```
pymc/
├── benchmarks/
├── docs/
├── pymc/
│   ├── __init__.py
│   ├── model.py
│   └── ...
├── scripts/
└── tests/
```

This is what is recommended by the [official Python packaging guide](https://packaging.python.org/en/latest/tutorials/packaging-projects/), by the [Pytest good practices guide](https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html#tests-outside-application-code), the [Poetry package management tool](https://python-poetry.org/docs/basic-usage/) as well as by [other guides](https://docs.python-guide.org/writing/structure/).
However, I have to note that `numpy`, `scipy`, and `pandas` all have `tests` inside the code directory.

Besides the arguments advanced by the [Pytest good practices guide](https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html#tests-outside-application-code), I can add that I often `grep` the source code directory to see where or how a function or class is used or defined. With the tests directory where it is currently, a lot of noise is often returned from the tests.

Cost of the transition: probably limited to some `import`s to change in the test files to reflect the move.

What is your opinion on this?</div>

### PR DESCRIPTION
**Pull request by Armavica**
Fixes #6526, as discussed in #6172.

**Checklist**
+ [ ] Explain important implementation details 👆
+ [ ] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [ ] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [ ] Are the changes covered by tests and docstrings?
+ [ ] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## New features
- ...

## Bugfixes
- ...

## Documentation
- ...

## Maintenance
- Move the tests outside of the code directory.

### CODE REVIEW

File: tests/backends/fixtures.py [hunk 1]
```diff
@@ -22,7 +22,8 @@
 import pytest
 
 from pymc.backends import base
-from pymc.tests import models
+
+from .. import models
```
Comment (ricardoV94) [inline tests/backends/fixtures.py]: You can still use absolute paths no? like `from tests import models`

That would be more readable.

**Comment by ricardoV94** [comment]
Don't know if we should label this as major, for those people who run tests when they install the package.

File: tests/backends/fixtures.py (hunk 1 above)
Comment (Armavica) [reply tests/backends/fixtures.py]: I was worried about `tests` conflicting with a Python package, but you are right, that should be possible.

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
A conflict seems to have cropped up, looks good to me otherwise

**Comment by ricardoV94** [comment]
@OriolAbril is the RTD failure new? It's not the usual timeout that was going on for a while

**Comment by OriolAbril** [comment]
The rtd error says it is due to an error when executing tge posterior predictive notebook, specifically in the pm.get_data line

**Comment by Armavica** [comment]
> The rtd error says it is due to an error when executing tge posterior predictive notebook, specifically in the pm.get_data line

Indeed, thank you! I found the mistake.

**Pull request merged**